hera_cal/vis_clean.py

```diff
@@ -7,29 +7,240 @@
 import datetime
 from uvtools import dspec
 import argparse
-
-
 from astropy import constants
 import copy
 import fnmatch
 from scipy import signal
+import warnings
+from pyuvdata import UVFlag
 
-from . import io
-from . import apply_cal
-from . import version
+from . import io, apply_cal, version, redcal
 from .datacontainer import DataContainer
 from .utils import echo
 from .flag_utils import factorize_flags
-import warnings
+
+
+def truncate_flagged_edges(data_in, weights_in, x, ax='freq'):
+    """
+    cut away edge channels and integrations that are completely flagged
+
+    Parameters
+    ----------
+    data_in : array-like, 2d
```

Comment:
also state here the implicit shape, i.e. `2d of shape (Ntimes, Nfreqs)`

Reply:
Added this.